server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -474,11 +448,10 @@ public void readFrom(StreamInput in) throws IOException {
             }
             mappings.put(type, source);
         }
-        int customSize = in.readVInt();
-        for (int i = 0; i < customSize; i++) {
-            String type = in.readString();
-            IndexMetaData.Custom customIndexMetaData = IndexMetaData.lookupPrototypeSafe(type).readFrom(in);
-            customs.put(type, customIndexMetaData);
+        if (in.getVersion().before(Version.V_7_0_0_alpha1)) {
+            // This used to be the size of custom metadata classes
+            int customSize = in.readVInt();
+            assert customSize == 0 : "unexpected custom metadata when none is supported";
```

Comment:
hmm do we need to skip the size if we are in production? I mean that assert will not trip if we run without -ea

Reply:
I'm not following, do you want it to be a regular exception instead of an assert? We still need to do the `readVInt` regardless of `-ea` or without, it's just the 0 size comparison that is an assert for tests

Reply:
I think that we need to do a hard exception indeed, we won't be able to read the rest of the message at this point because we can't read the customs (no `IndexMetaData#lookupPrototypeSafe anymore`) when there are unexpected ones. That is, `customSize > 0` is fatal for us.